# Using HuggingFace Transformers NLP Pipelines






In [ ]:
from huggingface_hub import login
login()

### 1. Zero-Shot Classifier

In [ ]:
from transformers import pipeline

classifier = pipeline("zero-shot-classification", model="facebook/bart-large-mnli")

output = classifier(
    "This is a course about the Transformers library. This course covers the popular Transformers library used for text analysis",
    candidate_labels=["education", "technology", "natural language processing", "machine learning"],
)
print("Classification Results:")
for label, score in zip(output['labels'], output['scores']):
    print(f"Label: {label} | Score: {score:.4f}")


/usr/local/lib/python3.10/dist-packages/torch_xla/__init__.py:253: UserWarning: `tensorflow` can conflict with `torch-xla`. Prefer `tensorflow-cpu` when using PyTorch/XLA. To silence this warning, `pip uninstall -y tensorflow && pip install tensorflow-cpu`. If you are in a notebook environment such as Colab or Kaggle, restart your notebook runtime afterwards.
  warnings.warn(
/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_token.py:89: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


config.json:   0%|          | 0.00/1.15k [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/1.63G [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/26.0 [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/899k [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/456k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/1.36M [00:00<?, ?B/s]

/usr/local/lib/python3.10/dist-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884
  warnings.warn(


Classification Results:
Label: technology | Score: 0.4869
Label: education | Score: 0.3116
Label: natural language processing | Score: 0.1433
Label: machine learning | Score: 0.0582


Zero-shot classification adalah metode dalam pemrosesan bahasa alami (NLP) yang memungkinkan model untuk mengklasifikasikan teks ke dalam kategori yang belum pernah dilatih sebelumnya.

> Pada kode diatas, zero-shot classification dijalankan menggunakan model `facebook/bart-large-mnli`.





Teks `"This is a course about the Transformers library. This course covers the popular Transformers library used for text analysis"` akan diklasifikasikan kedalam beberapa daftar label kandidat yang telah ditentukan yaitu `"education", "technology", "natural language processing", "machine learning"`.

Hasil menunjukkan bahwa teks tersebut mendapat skor confidence yang dinilai dari rentang 0-1 sebagai berikut :

1. technology: 0.4868
2. education: 0.3116
3. natural language processing: 0.1433
4. machine learning: 0.0582

Dari skor confidence diatas, dapat diketahui bahwa model yakin bahwa teks yang diberikan lebih masuk kedalam kategori `technology` daripada yang lainnya.

#### 1.1 Contoh yang lain

In [ ]:
from transformers import pipeline

model_name = "MoritzLaurer/mDeBERTa-v3-base-xnli-multilingual-nli-2mil7"
classifier = pipeline("zero-shot-classification", model=model_name)

output = classifier(
    "This is a course about the Transformers library. This course covers the popular Transformers library used for text analysis",
    candidate_labels=["education", "technology", "natural language processing", "machine learning"],
)

# Print the classification result in a structured format
print("Classification Results:")
for label, score in zip(output['labels'], output['scores']):
    print(f"Label: {label} | Score: {score:.4f}")


config.json:   0%|          | 0.00/1.09k [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/558M [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/467 [00:00<?, ?B/s]

spm.model:   0%|          | 0.00/4.31M [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/16.3M [00:00<?, ?B/s]

added_tokens.json:   0%|          | 0.00/23.0 [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/173 [00:00<?, ?B/s]

Classification Results:
Label: education | Score: 0.9483
Label: technology | Score: 0.0337
Label: machine learning | Score: 0.0147
Label: natural language processing | Score: 0.0032


In [ ]:
from transformers import pipeline

model_name = "cross-encoder/nli-MiniLM2-L6-H768"
classifier = pipeline("zero-shot-classification", model=model_name)

output = classifier(
    "This is a course about the Transformers library. This course covers the popular Transformers library used for text analysis",
    candidate_labels=["education", "technology", "natural language processing", "machine learning"],
)

# Print the classification result in a structured format
print("Classification Results:")
for label, score in zip(output['labels'], output['scores']):
    print(f"Label: {label} | Score: {score:.4f}")


config.json:   0%|          | 0.00/875 [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/329M [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/330 [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/798k [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/456k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/1.36M [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/239 [00:00<?, ?B/s]

Classification Results:
Label: education | Score: 0.6556
Label: technology | Score: 0.2615
Label: natural language processing | Score: 0.0487
Label: machine learning | Score: 0.0343


In [ ]:
from transformers import pipeline

model_name = "BSC-LT/sciroshot"
classifier = pipeline("zero-shot-classification", model=model_name)

output = classifier(
    "This is a course about the Transformers library. This course covers the popular Transformers library used for text analysis",
    candidate_labels=["education", "technology", "natural language processing", "machine learning"],
)

# Print the classification result in a structured format
print("Classification Results:")
for label, score in zip(output['labels'], output['scores']):
    print(f"Label: {label} | Score: {score:.4f}")

config.json:   0%|          | 0.00/865 [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/1.42G [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/295 [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/798k [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/456k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/1.36M [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/239 [00:00<?, ?B/s]

Classification Results:
Label: technology | Score: 0.2760
Label: natural language processing | Score: 0.2608
Label: education | Score: 0.2554
Label: machine learning | Score: 0.2078


### 2. Text Generator

In [ ]:
from transformers import pipeline

generator = pipeline("text-generation", model="gpt2")

output = generator(
    "In this Hugging Face's Transformers library course, we will teach you how to"
)

print("Generated Text:")
for i, generated in enumerate(output):
    print(f"Output {i+1}: {generated['generated_text']}")

config.json:   0%|          | 0.00/665 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/548M [00:00<?, ?B/s]

generation_config.json:   0%|          | 0.00/124 [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/26.0 [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/1.04M [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/456k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/1.36M [00:00<?, ?B/s]

Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.


Generated Text:
Output 1: In this Hugging Face's Transformers library course, we will teach you how to teach your own Optimus Prime to learn the Transformers abilities from an actual point-blank robot. The best example of this is a picture. For that reason, you should also


text-generation (pembuatan teks) adalah proses dalam NLP untuk menghasilkan teks baru berdasarkan input atau konteks yang diberikan.

> Pada kode diatas, text-generation dijalankan menggunakan model `gpt2`.



Dengan inputan seperti ini :

> In this Hugging Face's Transformers library course, we will teach you how to

Hasil yang ditunjukkan adalah seperti ini :
 > "In this Hugging Face's Transformers library course, we will teach you how to design a robot that can make up to 1 Billion cubic feet of water, using only one of the available materials. You will learn how to construct the robot by creating all"

Dalam hal ini, model belum secara sempurna mengerti konteks yang diberikan oleh input dan menghasilkan output yang berbeda dari pemikiran orang yang menginput.

In [ ]:
from transformers import pipeline

generator = pipeline("text-generation", model="gpt2")

output = generator(
    "I have a dream about getting a car that",
    max_length=50,
    num_return_sequences=2,
)

print("Generated Text:")
for i, generated in enumerate(output):
    print(f"Output {i+1}: {generated['generated_text']}")


Truncation was not explicitly activated but `max_length` is provided a specific value, please use `truncation=True` to explicitly truncate examples to max length. Defaulting to 'longest_first' truncation strategy. If you encode pairs of sequences (GLUE-style) with the tokenizer you can select this strategy more precisely by providing a specific strategy to `truncation`.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.


Generated Text:
Output 1: I have a dream about getting a car that is ready to go and not to have a car waiting. The more I try, the more I start to look for work in the parking lot. This is why I use the "No" tag.
Output 2: I have a dream about getting a car that makes me feel like a superstar and I cannot wait to drive up to Vegas!"

"Well now I'm ready… that's how the fans want to be," she said, pointing to the cars


Contoh diatas adalah penerapan lain untuk menjalankan `generator`.

Bagian `max_length` ditujukan untuk mengatur panjang maksimum teks yang dihasilkan.

> Pada Contoh diatas `max_length=50` artinya panjang maksimum kata yang keluar adalah 50 token atau kata.




Bagian `num_return_sequences` ditujukan untuk meminta model untuk menghasilkan dua hasil dari generator.

> Pada Contoh diatas `num_return_sequences=2` artinya meminta model untuk mengeluarkan 2 output teks.

#### 2.1 Contoh Lainnya

In [ ]:
from transformers import pipeline

model_name = "gpt2-medium"
generator = pipeline("text-generation", model=model_name)


output = generator(
    "I have a dream about getting a car that",
    max_length=50,
    num_return_sequences=2,
)

print("Generated Text:")
for i, generated in enumerate(output):
    print(f"Output {i+1}: {generated['generated_text']}")


config.json:   0%|          | 0.00/718 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/1.52G [00:00<?, ?B/s]

generation_config.json:   0%|          | 0.00/124 [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/26.0 [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/1.04M [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/456k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/1.36M [00:00<?, ?B/s]

Truncation was not explicitly activated but `max_length` is provided a specific value, please use `truncation=True` to explicitly truncate examples to max length. Defaulting to 'longest_first' truncation strategy. If you encode pairs of sequences (GLUE-style) with the tokenizer you can select this strategy more precisely by providing a specific strategy to `truncation`.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.


Generated Text:
Output 1: I have a dream about getting a car that can do 36 or 52 miles on the gas but doesn't need to be running 24 hours a day and is more sustainable than a car that needs oil. And it also would revolutionize clean energy."
Output 2: I have a dream about getting a car that actually drives itself. I'd like to ride along in a chauffeured limousine that drives itself." I'm still figuring it out, but these thoughts are good enough for me to continue on.


In [ ]:
from transformers import pipeline

model_name = "nvidia/Llama-3.1-Nemotron-70B-Instruct-HF"
generator = pipeline("text-generation", model=model_name)


output = generator(
    "I have a dream about getting a car that",
    max_length=50,
    num_return_sequences=2,
)

print("Generated Text:")
for i, generated in enumerate(output):
    print(f"Output {i+1}: {generated['generated_text']}")


config.json:   0%|          | 0.00/930 [00:00<?, ?B/s]

model.safetensors.index.json:   0%|          | 0.00/59.6k [00:00<?, ?B/s]

model-00001-of-00030.safetensors:   0%|          | 0.00/4.58G [00:00<?, ?B/s]

model-00002-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00003-of-00030.safetensors:   0%|          | 0.00/5.00G [00:00<?, ?B/s]

model-00004-of-00030.safetensors:   0%|          | 0.00/4.97G [00:00<?, ?B/s]

model-00005-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00006-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00007-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00008-of-00030.safetensors:   0%|          | 0.00/5.00G [00:00<?, ?B/s]

model-00009-of-00030.safetensors:   0%|          | 0.00/4.97G [00:00<?, ?B/s]

model-00010-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00011-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00012-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00013-of-00030.safetensors:   0%|          | 0.00/5.00G [00:00<?, ?B/s]

model-00014-of-00030.safetensors:   0%|          | 0.00/4.97G [00:00<?, ?B/s]

model-00015-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00016-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00017-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00018-of-00030.safetensors:   0%|          | 0.00/5.00G [00:00<?, ?B/s]

model-00019-of-00030.safetensors:   0%|          | 0.00/4.97G [00:00<?, ?B/s]

model-00020-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00021-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00022-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00023-of-00030.safetensors:   0%|          | 0.00/5.00G [00:00<?, ?B/s]

model-00024-of-00030.safetensors:   0%|          | 0.00/4.97G [00:00<?, ?B/s]

model-00025-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00026-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00027-of-00030.safetensors:   0%|          | 0.00/4.66G [00:00<?, ?B/s]

model-00028-of-00030.safetensors:   0%|          | 0.00/5.00G [00:00<?, ?B/s]

model-00029-of-00030.safetensors:   0%|          | 0.00/4.97G [00:00<?, ?B/s]

model-00030-of-00030.safetensors:   0%|          | 0.00/2.10G [00:00<?, ?B/s]

Loading checkpoint shards:   0%|          | 0/30 [00:00<?, ?it/s]

generation_config.json:   0%|          | 0.00/163 [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/55.2k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/9.09M [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/296 [00:00<?, ?B/s]

Truncation was not explicitly activated but `max_length` is provided a specific value, please use `truncation=True` to explicitly truncate examples to max length. Defaulting to 'longest_first' truncation strategy. If you encode pairs of sequences (GLUE-style) with the tokenizer you can select this strategy more precisely by providing a specific strategy to `truncation`.


ValueError: Greedy methods without beam search do not support `num_return_sequences` different than 1 (got 2).

In [ ]:
from transformers import pipeline

model_name = "shuttleai/shuttle-3"
generator = pipeline("text-generation", model=model_name)


output = generator(
    "I have a dream about getting a car that",
    max_length=50,
    num_return_sequences=2,
)

print("Generated Text:")
for i, generated in enumerate(output):
    print(f"Output {i+1}: {generated['generated_text']}")


In [ ]:
from transformers import pipeline

model_name = "microsoft/Phi-3.5-mini-instruct"
generator = pipeline("text-generation", model=model_name)


output = generator(
    "I have a dream about getting a car that",
    max_length=50,
    num_return_sequences=2,
)

print("Generated Text:")
for i, generated in enumerate(output):
    print(f"Output {i+1}: {generated['generated_text']}")


### 3. Fill-Mask

In [ ]:
from transformers import pipeline

unmasker = pipeline("fill-mask")
input_sentence = "I want to make me a beautiful <mask> to feeling good about myself."
output = unmasker(input_sentence, top_k=2)

print("Predicted Words for the Mask:")
for i, prediction in enumerate(output):
    predicted_word = prediction['token_str']
    score = prediction['score']
    full_sentence = input_sentence.replace("<mask>", predicted_word)

    print(f"Prediction {i+1}: {predicted_word} (Score: {score:.4f})")
    print(f"Full Sentence: {full_sentence}\n")


No model was supplied, defaulted to distilbert/distilroberta-base and revision ec58a5b (https://huggingface.co/distilbert/distilroberta-base).
Using a pipeline without specifying a model name and revision in production is not recommended.
Some weights of the model checkpoint at distilbert/distilroberta-base were not used when initializing RobertaForMaskedLM: ['roberta.pooler.dense.bias', 'roberta.pooler.dense.weight']
- This IS expected if you are initializing RobertaForMaskedLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing RobertaForMaskedLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).


Predicted Words for the Mask:
Prediction 1:  guide (Score: 0.1228)
Full Sentence: I want to make me a beautiful  guide to feeling good about myself.

Prediction 2:  monument (Score: 0.1146)
Full Sentence: I want to make me a beautiful  monument to feeling good about myself.



Fill-mask adalah sebuah teknik dalam pemrosesan bahasa alami (NLP) yang digunakan untuk mengisi kata yang hilang dalam kalimat.

Dengan inputan seperti ini :

> "I want to make me a beautiful < mask > to feeling good about myself."

Token `<mask>` adalah kata yang hilang dan perlu diisi oleh model

Hasil yang ditunjukkan adalah seperti ini :
 > 1. I want to make me a beautiful guide to feeling good about myself.

 > 2. I want to make me a beautiful monument to feeling good about myself.

Dari hasil ke 1, model menggunakan kata `guide` untuk mengisi mask dengan skor confidence 0.1227. Sedangkan untuk kata `monument`, model mengisinya dengan skor confidence 0.1146

#### 3.1 Contoh Lainnya

In [ ]:
from transformers import pipeline

model_name = "google-bert/bert-base-uncased"

unmasker = pipeline("fill-mask")
input_sentence = "I want to make me a beautiful <mask> to feeling good about myself."
output = unmasker(input_sentence, top_k=2)

print("Predicted Words for the Mask:")
for i, prediction in enumerate(output):
    predicted_word = prediction['token_str']
    score = prediction['score']
    full_sentence = input_sentence.replace("<mask>", predicted_word)

    print(f"Prediction {i+1}: {predicted_word} (Score: {score:.4f})")
    print(f"Full Sentence: {full_sentence}\n")



No model was supplied, defaulted to distilbert/distilroberta-base and revision ec58a5b (https://huggingface.co/distilbert/distilroberta-base).
Using a pipeline without specifying a model name and revision in production is not recommended.
Some weights of the model checkpoint at distilbert/distilroberta-base were not used when initializing RobertaForMaskedLM: ['roberta.pooler.dense.bias', 'roberta.pooler.dense.weight']
- This IS expected if you are initializing RobertaForMaskedLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing RobertaForMaskedLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).


Predicted Words for the Mask:
Prediction 1:  guide (Score: 0.1228)
Full Sentence: I want to make me a beautiful  guide to feeling good about myself.

Prediction 2:  monument (Score: 0.1146)
Full Sentence: I want to make me a beautiful  monument to feeling good about myself.



In [ ]:
from transformers import pipeline

model_name = "distilbert/distilbert-base-uncased"

unmasker = pipeline("fill-mask")
input_sentence = "I want to make me a beautiful <mask> to feeling good about myself."
output = unmasker(input_sentence, top_k=2)

print("Predicted Words for the Mask:")
for i, prediction in enumerate(output):
    predicted_word = prediction['token_str']
    score = prediction['score']
    full_sentence = input_sentence.replace("<mask>", predicted_word)

    print(f"Prediction {i+1}: {predicted_word} (Score: {score:.4f})")
    print(f"Full Sentence: {full_sentence}\n")



No model was supplied, defaulted to distilbert/distilroberta-base and revision ec58a5b (https://huggingface.co/distilbert/distilroberta-base).
Using a pipeline without specifying a model name and revision in production is not recommended.
Some weights of the model checkpoint at distilbert/distilroberta-base were not used when initializing RobertaForMaskedLM: ['roberta.pooler.dense.bias', 'roberta.pooler.dense.weight']
- This IS expected if you are initializing RobertaForMaskedLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing RobertaForMaskedLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).


Predicted Words for the Mask:
Prediction 1:  guide (Score: 0.1228)
Full Sentence: I want to make me a beautiful  guide to feeling good about myself.

Prediction 2:  monument (Score: 0.1146)
Full Sentence: I want to make me a beautiful  monument to feeling good about myself.



In [ ]:
from transformers import pipeline

model_name = "FacebookAI/roberta-base"

unmasker = pipeline("fill-mask")
input_sentence = "I want to make me a beautiful <mask> to feeling good about myself."
output = unmasker(input_sentence, top_k=2)

print("Predicted Words for the Mask:")
for i, prediction in enumerate(output):
    predicted_word = prediction['token_str']
    score = prediction['score']
    full_sentence = input_sentence.replace("<mask>", predicted_word)

    print(f"Prediction {i+1}: {predicted_word} (Score: {score:.4f})")
    print(f"Full Sentence: {full_sentence}\n")



No model was supplied, defaulted to distilbert/distilroberta-base and revision ec58a5b (https://huggingface.co/distilbert/distilroberta-base).
Using a pipeline without specifying a model name and revision in production is not recommended.
Some weights of the model checkpoint at distilbert/distilroberta-base were not used when initializing RobertaForMaskedLM: ['roberta.pooler.dense.bias', 'roberta.pooler.dense.weight']
- This IS expected if you are initializing RobertaForMaskedLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing RobertaForMaskedLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).


Predicted Words for the Mask:
Prediction 1:  guide (Score: 0.1228)
Full Sentence: I want to make me a beautiful  guide to feeling good about myself.

Prediction 2:  monument (Score: 0.1146)
Full Sentence: I want to make me a beautiful  monument to feeling good about myself.



In [ ]:
from transformers import pipeline

model_name = "microsoft/mdeberta-v3-base"

unmasker = pipeline("fill-mask")
input_sentence = "I want to make me a beautiful <mask> to feeling good about myself."
output = unmasker(input_sentence, top_k=2)

print("Predicted Words for the Mask:")
for i, prediction in enumerate(output):
    predicted_word = prediction['token_str']
    score = prediction['score']
    full_sentence = input_sentence.replace("<mask>", predicted_word)

    print(f"Prediction {i+1}: {predicted_word} (Score: {score:.4f})")
    print(f"Full Sentence: {full_sentence}\n")



No model was supplied, defaulted to distilbert/distilroberta-base and revision ec58a5b (https://huggingface.co/distilbert/distilroberta-base).
Using a pipeline without specifying a model name and revision in production is not recommended.
Some weights of the model checkpoint at distilbert/distilroberta-base were not used when initializing RobertaForMaskedLM: ['roberta.pooler.dense.bias', 'roberta.pooler.dense.weight']
- This IS expected if you are initializing RobertaForMaskedLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing RobertaForMaskedLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).


Predicted Words for the Mask:
Prediction 1:  guide (Score: 0.1228)
Full Sentence: I want to make me a beautiful  guide to feeling good about myself.

Prediction 2:  monument (Score: 0.1146)
Full Sentence: I want to make me a beautiful  monument to feeling good about myself.



### 4. Named Entity Recognition (NER)

In [ ]:
from transformers import pipeline

ner = pipeline("ner", grouped_entities=True)
text = "My Name Is Muhammad Ivan Aldorino. I am a student from Universitas Negeri Semarang and now I attended MSIB BATCH 7 at Infinite Learning. I think Semarang is a great place to take a degree on."
output = ner(text)

print("Named Entities:")
for entity in output:
    entity_str = entity['word']
    entity_type = entity['entity_group']
    start = entity['start']
    end = entity['end']

    print(f"Entity: {entity_str} | Type: {entity_type} | Position: ({start}, {end})")


No model was supplied, defaulted to dbmdz/bert-large-cased-finetuned-conll03-english and revision f2482bf (https://huggingface.co/dbmdz/bert-large-cased-finetuned-conll03-english).
Using a pipeline without specifying a model name and revision in production is not recommended.
Some weights of the model checkpoint at dbmdz/bert-large-cased-finetuned-conll03-english were not used when initializing BertForTokenClassification: ['bert.pooler.dense.bias', 'bert.pooler.dense.weight']
- This IS expected if you are initializing BertForTokenClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForTokenClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
/usr/local/lib/python3.10/dist-p

Named Entities:
Entity: Muhammad Ivan Aldorino | Type: PER | Position: (11, 33)
Entity: Universitas Negeri Semarang | Type: ORG | Position: (55, 82)
Entity: MS | Type: ORG | Position: (102, 104)
Entity: Infinite Learning | Type: ORG | Position: (118, 135)
Entity: Semarang | Type: LOC | Position: (145, 153)


Named Entity Recognition (NER) adalah sebuah teknik dalam pemrosesan bahasa alami (NLP) yang digunakan untuk mengidentifikasi dan mengklasifikasikan entity bernama dalam teks.

`grouped_entities=True` menunjukkan bahwa hasil pengenalan entities akan dikelompokkan untuk mempermudah membacanya.



Hasil dari identifikasi oleh model adalah :


> 1. Entity Pertama</br>
- entity_group: PER (Person)
- score: 0.998613
- word: "Muhammad Ivan Aldorino"
- start: 11
- end: 33</br></br>
artinya, model mengenali kata "Muhammad Ivan Aldorino" sebagai nama orang dan memasukkannya kedalam entity group `PER` dengan skor confidence 0.9986

> 2. Entity Kedua</br>
- entity_group: ORG (Organization)
- score: 0.9639477
- word: "Universitas Negeri Semarang"
- start: 56
- end: 83</br></br>
artinya, model mengenali kata "Universitas Negeri Semarang" sebagai nama organisasi dan memasukkannya kedalam entity group `ORG` dengan skor confidence 0.9639


> 3. Entity Ketiga</br>
- entity_group: ORG (Organization)
- score: 0.95237684
- word: "Infinite Learning"
- start: 120
- end: 137</br></br>
artinya, model mengenali kata "Infinite Learning" sebagai nama organisasi dan memasukkannya kedalam entity group `ORG` dengan skor confidence 0.9523

> 4. Entity Keempat</br>
- entity_group: LOC (Location)
- score: 0.9933
- word: "Semarang"
- start: 147
- end: 155</br></br>
artinya, model mengenali kata "semarang" sebagai nama organisasi dan memasukkannya kedalam entity group `LOC` dengan skor confidence 0.9933


#### 4.1 Contoh Lainnya

In [ ]:
from transformers import pipeline

model_name = "Babelscape/wikineural-multilingual-ner"
ner = pipeline("ner", grouped_entities=True, model=model_name)
text = "My Name Is Muhammad Ivan Aldorino. I am a student from Universitas Negeri Semarang and now I attended MSIB BATCH 7 at Infinite Learning. I think Semarang is a great place to take a degree on."
output = ner(text)

print("Named Entities:")
for entity in output:
    entity_str = entity['word']
    entity_type = entity['entity_group']
    start = entity['start']
    end = entity['end']

    print(f"Entity: {entity_str} | Type: {entity_type} | Position: ({start}, {end})")


Named Entities:
Entity: Muhammad Ivan Aldorino | Type: PER | Position: (11, 33)
Entity: Universitas Negeri Semarang | Type: ORG | Position: (55, 82)
Entity: Infinite Learning | Type: MISC | Position: (118, 135)
Entity: Semarang | Type: LOC | Position: (145, 153)


In [ ]:
from transformers import pipeline

model_name = "pysentimiento/robertuito-ner"
ner = pipeline("ner", grouped_entities=True, model=model_name)
text = "My Name Is Muhammad Ivan Aldorino. I am a student from Universitas Negeri Semarang and now I attended MSIB BATCH 7 at Infinite Learning. I think Semarang is a great place to take a degree on."
output = ner(text)

print("Named Entities:")
for entity in output:
    entity_str = entity['word']
    entity_type = entity['entity_group']
    start = entity['start']
    end = entity['end']

    print(f"Entity: {entity_str} | Type: {entity_type} | Position: ({start}, {end})")


Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.


Named Entities:
Entity: muhammad ivan aldorino. | Type: PER | Position: (10, 34)
Entity: universitas | Type: ORG | Position: (54, 66)
Entity: negeri semarang | Type: LOC | Position: (66, 82)
Entity: m | Type: ORG | Position: (101, 103)
Entity: si | Type: OTHER | Position: (103, 105)
Entity: b | Type: PROD | Position: (105, 106)
Entity: tch 7 | Type: PROD | Position: (109, 114)
Entity: inf | Type: ORG | Position: (117, 121)
Entity: inite learning | Type: PROD | Position: (121, 135)
Entity: semarang | Type: LOC | Position: (144, 153)


### 5. Question Answering

In [ ]:
from transformers import pipeline

qa_pipeline = pipeline("question-answering", model="distilbert-base-cased-distilled-squad")
context = "Semarang terkenal akan Bandeng Presto sehingga sering dijadikan opsi oleh-oleh bagi para turis."
question = "Kota mana yang terkenal dengan Bandeng Presto?"

result = qa_pipeline(question=question, context=context)
print(f"Jawaban: {result['answer']}")

config.json:   0%|          | 0.00/473 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/261M [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/49.0 [00:00<?, ?B/s]

vocab.txt:   0%|          | 0.00/213k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/436k [00:00<?, ?B/s]

Jawaban: Semarang


Question-Answering (QA) adalah teknik dalam NLP yang fokus pada kemampuan suatu sistem untuk menjawab pertanyaan berdasarkan informasi yang diberikan.



Dalam question-answering, dibutuhkan sebuah konteks agar model dapat menjawab pertanyaan yang diberikan.

`context` adalah sebuah konteks yang diberikan kepada NLP terkait informasi tertentu. Setelah diberikan konteks, maka pertanyaan dapat dimasukkan pada `question` yang sesuai dengan konteksnya.


Sebagai contoh, akan dimasukkan `context` seperti ini : </br>
"Semarang terkenal akan Bandeng Presto sehingga sering dijadikan opsi oleh-oleh bagi para turis."
</br></br>
lalu akan diberikan `question` seperti ini :</br>
Kota mana yang terkenal dengan Bandeng Presto?


Setelah code dijalankan, akan keluar `answer` dari sebuah `question` yang dianalisa menggunakan `context` yang ada.

Jawaban yang keluar adalah "Semarang"

#### 5.1 Contoh Lainnya

In [ ]:
from transformers import pipeline

model_name = "deepset/roberta-base-squad2"

qa_pipeline = pipeline("question-answering", model=model_name)
context = "Semarang terkenal akan Bandeng Presto sehingga sering dijadikan opsi oleh-oleh bagi para turis."
question = "Kota mana yang terkenal dengan Bandeng Presto?"

result = qa_pipeline(question=question, context=context)
print(f"Jawaban: {result['answer']}")

Jawaban: Semarang


In [ ]:
from transformers import pipeline

model_name = "distilbert/distilbert-base-cased-distilled-squad"

qa_pipeline = pipeline("question-answering", model=model_name)
context = "Semarang terkenal akan Bandeng Presto sehingga sering dijadikan opsi oleh-oleh bagi para turis."
question = "Kota mana yang terkenal dengan Bandeng Presto?"

result = qa_pipeline(question=question, context=context)
print(f"Jawaban: {result['answer']}")

Jawaban: Semarang


In [ ]:
from transformers import pipeline

model_name = "Intel/dynamic_tinybert"

qa_pipeline = pipeline("question-answering", model=model_name)
context = "Semarang terkenal akan Bandeng Presto sehingga sering dijadikan opsi oleh-oleh bagi para turis."
question = "Kota mana yang terkenal dengan Bandeng Presto?"

result = qa_pipeline(question=question, context=context)
print(f"Jawaban: {result['answer']}")

Jawaban: sehingga


### 6. Sentiment Analysis

In [ ]:
from transformers import pipeline


classifier = pipeline("sentiment-analysis")


text = "Nasi padang and fried chicken are the perfect combination, and when someone buys it for me, I feel so delighted!"
output = classifier(text)

print("Sentiment Analysis Result:")
for result in output:
    label = result['label']
    score = result['score']
    print(f"Sentiment: {label} | Score: {score:.4f}")


No model was supplied, defaulted to distilbert/distilbert-base-uncased-finetuned-sst-2-english and revision af0f99b (https://huggingface.co/distilbert/distilbert-base-uncased-finetuned-sst-2-english).
Using a pipeline without specifying a model name and revision in production is not recommended.


Sentiment Analysis Result:
Sentiment: POSITIVE | Score: 0.9998


Sentiment Analysis merupakan salah satu teknik dalam NLP untuk mengidentifikasi dan mengklasifikasikan emosi atau opini yang terkandung dalam teks.


Sebagai Contoh, saya menuliskan kata "Nasi padang and fried chicken are the perfect combination, and when someone buys it for me, I feel so delighted!".

Teks tersebut kemudian diterima oleh model dan dilakukan analisa sentimen untuk menentukan apakah harus dilabeli POSITIVE atau NEGATIVE.



Sebagai contoh, kata diatas diklasifikasi oleh model sebagai kata yang `POSITIVE` dan memiliki skor confidence yaitu 0.9997

In [ ]:
from transformers import pipeline


classifier = pipeline("sentiment-analysis")


text = "Matcha and Chocolate are the worst drink combination ever existed. I wonder how people can drink such a thing!"
output = classifier(text)

print("Sentiment Analysis Result:")
for result in output:
    label = result['label']
    score = result['score']
    print(f"Sentiment: {label} | Score: {score:.4f}")


No model was supplied, defaulted to distilbert/distilbert-base-uncased-finetuned-sst-2-english and revision af0f99b (https://huggingface.co/distilbert/distilbert-base-uncased-finetuned-sst-2-english).
Using a pipeline without specifying a model name and revision in production is not recommended.


Sentiment Analysis Result:
Sentiment: NEGATIVE | Score: 0.9998


Diatas merupakan contoh ketika model menganalisis sentimen yang ada didalam kata dan melabelinya menjadi `NEGATIVE` dengan skor confidence yaitu 0.9998

#### 6.1 Contoh Lainnya

In [ ]:
from transformers import pipeline

model_name = "ProsusAI/finbert"

text = "Nasi padang and fried chicken are the perfect combination, and when someone buys it for me, I feel so delighted!"
output = classifier(text)

print("Sentiment Analysis Result:")
for result in output:
    label = result['label']
    score = result['score']
    print(f"Sentiment: {label} | Score: {score:.4f}")

Sentiment Analysis Result:
Sentiment: POSITIVE | Score: 0.9998


In [ ]:
from transformers import pipeline

model_name = "ProsusAI/finbert"

text = "Matcha and Chocolate are the worst drink combination ever existed. I wonder how people can drink such a thing!"
output = classifier(text)

print("Sentiment Analysis Result:")
for result in output:
    label = result['label']
    score = result['score']
    print(f"Sentiment: {label} | Score: {score:.4f}")

Sentiment Analysis Result:
Sentiment: NEGATIVE | Score: 0.9998


In [ ]:
from transformers import pipeline

model_name = "nlptown/bert-base-multilingual-uncased-sentiment"

text = "Nasi padang and fried chicken are the perfect combination, and when someone buys it for me, I feel so delighted!"
output = classifier(text)

print("Sentiment Analysis Result:")
for result in output:
    label = result['label']
    score = result['score']
    print(f"Sentiment: {label} | Score: {score:.4f}")


Sentiment Analysis Result:
Sentiment: POSITIVE | Score: 0.9998


In [ ]:
from transformers import pipeline

model_name = "nlptown/bert-base-multilingual-uncased-sentiment"

text = "Matcha and Chocolate are the worst drink combination ever existed. I wonder how people can drink such a thing!"
output = classifier(text)

print("Sentiment Analysis Result:")
for result in output:
    label = result['label']
    score = result['score']
    print(f"Sentiment: {label} | Score: {score:.4f}")

Sentiment Analysis Result:
Sentiment: NEGATIVE | Score: 0.9998


In [ ]:
from transformers import pipeline

model_name = "tezign/Erlangshen-Sentiment-FineTune"

text = "Nasi padang and fried chicken are the perfect combination, and when someone buys it for me, I feel so delighted!"
output = classifier(text)

print("Sentiment Analysis Result:")
for result in output:
    label = result['label']
    score = result['score']
    print(f"Sentiment: {label} | Score: {score:.4f}")

Sentiment Analysis Result:
Sentiment: POSITIVE | Score: 0.9998


In [ ]:
from transformers import pipeline

model_name = "tezign/Erlangshen-Sentiment-FineTune"

text = "Matcha and Chocolate are the worst drink combination ever existed. I wonder how people can drink such a thing!"
output = classifier(text)

print("Sentiment Analysis Result:")
for result in output:
    label = result['label']
    score = result['score']
    print(f"Sentiment: {label} | Score: {score:.4f}")

Sentiment Analysis Result:
Sentiment: NEGATIVE | Score: 0.9998


### 7. Summarization

In [ ]:
from transformers import pipeline

summarizer = pipeline("summarization")
summarizer(
    """
    Semarang, the capital of Central Java province in Indonesia, is a vibrant city that combines rich
    history with modern development. Situated on the north coast of Java, it is known for its diverse
    culture and heritage, which reflect a blend of Javanese, Chinese, and Dutch influences.

    One of Semarang's most iconic landmarks is the Lawang Sewu, a historical building that once served
    as the headquarters of the Dutch East Indies Railway Company. Its name, meaning "Thousand Doors,"
    refers to the numerous doors and windows that adorn the structure. Another notable site is the Sam
    Poo Kong Temple, dedicated to the Chinese explorer Zheng He, showcasing the city's multicultural
    roots.

    The city's culinary scene is also noteworthy, offering a variety of local dishes such as lumpia
    Semarang (spring rolls), tahu gimbal (tofu with shrimp fritters), and bandeng presto (fried milkfish).
    Semarang is also a gateway to several natural attractions, including the nearby Dieng Plateau and the
    scenic beach at Teluk Awur.

    With its unique blend of history, culture, and cuisine, Semarang is a captivating destination for
    both local and international tourists.
"""
)

No model was supplied, defaulted to sshleifer/distilbart-cnn-12-6 and revision a4f8f3e (https://huggingface.co/sshleifer/distilbart-cnn-12-6).
Using a pipeline without specifying a model name and revision in production is not recommended.


config.json:   0%|          | 0.00/1.80k [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/1.22G [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/26.0 [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/899k [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/456k [00:00<?, ?B/s]

[{'summary_text': " Semarang is the capital of Central Java province in Indonesia . Situated on the north coast of Java, it is known for its diverse culture and heritage . The Lawang Sewu is the headquarters of the Dutch East Indies Railway Company . The city's culinary scene is also noteworthy, offering a variety of local dishes such as spring rolls ."}]

Summarization adalah salah satu proses dalam NLP untuk merangkum input yang diberikan tanpa kehilangan inti informasi.

Input yang berisi teks dimasukkan ke dalam `summarizer` yang nanti akan diolah oleh model untuk dijadikan rangkuman yang lebih sedikit isinya tanpa kehilangan inti isinya.


#### 7.1 Contoh yang lain

[Contoh Model Disini](https://huggingface.co/models?pipeline_tag=summarization)

In [ ]:
from transformers import pipeline

model_name = "utrobinmv/t5_summary_en_ru_zh_base_2048"

summarizer = pipeline("summarization", model=model_name)
summarizer(
    """
    Semarang, the capital of Central Java province in Indonesia, is a vibrant city that combines rich
    history with modern development. Situated on the north coast of Java, it is known for its diverse
    culture and heritage, which reflect a blend of Javanese, Chinese, and Dutch influences.

    One of Semarang's most iconic landmarks is the Lawang Sewu, a historical building that once served
    as the headquarters of the Dutch East Indies Railway Company. Its name, meaning "Thousand Doors,"
    refers to the numerous doors and windows that adorn the structure. Another notable site is the Sam
    Poo Kong Temple, dedicated to the Chinese explorer Zheng He, showcasing the city's multicultural
    roots.

    The city's culinary scene is also noteworthy, offering a variety of local dishes such as lumpia
    Semarang (spring rolls), tahu gimbal (tofu with shrimp fritters), and bandeng presto (fried milkfish).
    Semarang is also a gateway to several natural attractions, including the nearby Dieng Plateau and the
    scenic beach at Teluk Awur.

    With its unique blend of history, culture, and cuisine, Semarang is a captivating destination for
    both local and international tourists.
"""
)

config.json:   0%|          | 0.00/809 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/1.19G [00:00<?, ?B/s]

generation_config.json:   0%|          | 0.00/206 [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/20.6k [00:00<?, ?B/s]

spiece.model:   0%|          | 0.00/1.47M [00:00<?, ?B/s]

added_tokens.json:   0%|          | 0.00/2.59k [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/2.54k [00:00<?, ?B/s]

[{'summary_text': "In our series of letters from African journalists, Zeinab Mohammed Salih looks at Semarang, one of Indonesia's most famous cities."}]

In [ ]:
from transformers import pipeline

model_name = "pszemraj/led-large-book-summary"

summarizer = pipeline("summarization", model=model_name)
summarizer(
    """
    Semarang, the capital of Central Java province in Indonesia, is a vibrant city that combines rich
    history with modern development. Situated on the north coast of Java, it is known for its diverse
    culture and heritage, which reflect a blend of Javanese, Chinese, and Dutch influences.

    One of Semarang's most iconic landmarks is the Lawang Sewu, a historical building that once served
    as the headquarters of the Dutch East Indies Railway Company. Its name, meaning "Thousand Doors,"
    refers to the numerous doors and windows that adorn the structure. Another notable site is the Sam
    Poo Kong Temple, dedicated to the Chinese explorer Zheng He, showcasing the city's multicultural
    roots.

    The city's culinary scene is also noteworthy, offering a variety of local dishes such as lumpia
    Semarang (spring rolls), tahu gimbal (tofu with shrimp fritters), and bandeng presto (fried milkfish).
    Semarang is also a gateway to several natural attractions, including the nearby Dieng Plateau and the
    scenic beach at Teluk Awur.

    With its unique blend of history, culture, and cuisine, Semarang is a captivating destination for
    both local and international tourists.
"""
)

config.json:   0%|          | 0.00/1.44k [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/1.84G [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/1.32k [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/798k [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/456k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/2.11M [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/772 [00:00<?, ?B/s]

Your max_length is set to 1024, but your input_length is only 305. Since this is a summarization task, where outputs shorter than the input are typically wanted, you might consider decreasing max_length manually, e.g. summarizer('...', max_length=152)
Input ids are automatically padded from 305 to 1024 to be a multiple of `config.attention_window`: 1024


[{'summary_text': 'This chapter opens with a description of Semarang, the capital of Central Java province in Indonesia. It is a "vibrant city that combines rich history with modern development" . Known for its diverse culture and heritage, it is a blend of Javanese, Chinese, and Dutch influences, the city contains important landmarks such as the Lawang Sewu, the Temple of the Chinese explorer Zheng He, and the Dieng Plateau, as well as the scenic beach at Teluk Awur. The narrator assures the reader that this chapter will be a must-see for both local and international tourists.'}]

In [ ]:
from transformers import pipeline

model_name = "google/bigbird-pegasus-large-pubmed"

summarizer = pipeline("summarization", model=model_name)
summarizer(
    """
    Semarang, the capital of Central Java province in Indonesia, is a vibrant city that combines rich
    history with modern development. Situated on the north coast of Java, it is known for its diverse
    culture and heritage, which reflect a blend of Javanese, Chinese, and Dutch influences.

    One of Semarang's most iconic landmarks is the Lawang Sewu, a historical building that once served
    as the headquarters of the Dutch East Indies Railway Company. Its name, meaning "Thousand Doors,"
    refers to the numerous doors and windows that adorn the structure. Another notable site is the Sam
    Poo Kong Temple, dedicated to the Chinese explorer Zheng He, showcasing the city's multicultural
    roots.

    The city's culinary scene is also noteworthy, offering a variety of local dishes such as lumpia
    Semarang (spring rolls), tahu gimbal (tofu with shrimp fritters), and bandeng presto (fried milkfish).
    Semarang is also a gateway to several natural attractions, including the nearby Dieng Plateau and the
    scenic beach at Teluk Awur.

    With its unique blend of history, culture, and cuisine, Semarang is a captivating destination for
    both local and international tourists.
"""
)

config.json:   0%|          | 0.00/1.05k [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/2.31G [00:00<?, ?B/s]

generation_config.json:   0%|          | 0.00/232 [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/1.20k [00:00<?, ?B/s]

spiece.model:   0%|          | 0.00/1.92M [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/3.51M [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/775 [00:00<?, ?B/s]

Your max_length is set to 256, but your input_length is only 236. Since this is a summarization task, where outputs shorter than the input are typically wanted, you might consider decreasing max_length manually, e.g. summarizer('...', max_length=118)
Attention type 'block_sparse' is not possible if sequence_length: 236 <= num global tokens: 2 * config.block_size + min. num sliding tokens: 3 * config.block_size + config.num_random_blocks * config.block_size + additional buffer: config.num_random_blocks * config.block_size = 704 with config.block_size = 64, config.num_random_blocks = 3. Changing attention type to 'original_full'...


[{'summary_text': 'despite the recent declines in environmental lead exposure in the united states and mexico , environmental lead exposure continues to be a significant public health problem .<n> environmental lead exposure is a leading cause of death and non - accidental morbidity in children , adolescents , and young adults .<n> mexico has the highest per capita level of environmental lead in the world .<n> however , the magnitude of lead exposure in the united states is much less than in mexico . in this article , we examine the relationship between the level of lead exposure in the united states and mexico , the relationship between lead levels in the united states and mexico , and the relationship between lead levels in the united states and mexico .<n> the article begins with an overview of the characteristics that make the united states and mexico different from other countries .<n> then , based on the characteristics that make the united states and mexico different from other 

In [ ]:
from transformers import pipeline

model_name = "human-centered-summarization/financial-summarization-pegasus"

summarizer = pipeline("summarization", model=model_name)
summarizer(
    """
    Semarang, the capital of Central Java province in Indonesia, is a vibrant city that combines rich
    history with modern development. Situated on the north coast of Java, it is known for its diverse
    culture and heritage, which reflect a blend of Javanese, Chinese, and Dutch influences.

    One of Semarang's most iconic landmarks is the Lawang Sewu, a historical building that once served
    as the headquarters of the Dutch East Indies Railway Company. Its name, meaning "Thousand Doors,"
    refers to the numerous doors and windows that adorn the structure. Another notable site is the Sam
    Poo Kong Temple, dedicated to the Chinese explorer Zheng He, showcasing the city's multicultural
    roots.

    The city's culinary scene is also noteworthy, offering a variety of local dishes such as lumpia
    Semarang (spring rolls), tahu gimbal (tofu with shrimp fritters), and bandeng presto (fried milkfish).
    Semarang is also a gateway to several natural attractions, including the nearby Dieng Plateau and the
    scenic beach at Teluk Awur.

    With its unique blend of history, culture, and cuisine, Semarang is a captivating destination for
    both local and international tourists.
"""
)

config.json:   0%|          | 0.00/1.27k [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/2.28G [00:00<?, ?B/s]

Some weights of PegasusForConditionalGeneration were not initialized from the model checkpoint at human-centered-summarization/financial-summarization-pegasus and are newly initialized: ['model.decoder.embed_positions.weight', 'model.encoder.embed_positions.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


tokenizer_config.json:   0%|          | 0.00/1.44k [00:00<?, ?B/s]

spiece.model:   0%|          | 0.00/1.91M [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/1.34k [00:00<?, ?B/s]

[{'summary_text': 'Situated on the north coast of Java, it is known for its diverse culture and heritage.'}]

In [ ]:
from transformers import pipeline

model_name = "toloka/t5-large-for-text-aggregation"

summarizer = pipeline("summarization", model=model_name)
summarizer(
    """
    Semarang, the capital of Central Java province in Indonesia, is a vibrant city that combines rich
    history with modern development. Situated on the north coast of Java, it is known for its diverse
    culture and heritage, which reflect a blend of Javanese, Chinese, and Dutch influences.

    One of Semarang's most iconic landmarks is the Lawang Sewu, a historical building that once served
    as the headquarters of the Dutch East Indies Railway Company. Its name, meaning "Thousand Doors,"
    refers to the numerous doors and windows that adorn the structure. Another notable site is the Sam
    Poo Kong Temple, dedicated to the Chinese explorer Zheng He, showcasing the city's multicultural
    roots.

    The city's culinary scene is also noteworthy, offering a variety of local dishes such as lumpia
    Semarang (spring rolls), tahu gimbal (tofu with shrimp fritters), and bandeng presto (fried milkfish).
    Semarang is also a gateway to several natural attractions, including the nearby Dieng Plateau and the
    scenic beach at Teluk Awur.

    With its unique blend of history, culture, and cuisine, Semarang is a captivating destination for
    both local and international tourists.
"""
)

config.json:   0%|          | 0.00/1.42k [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/2.95G [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/1.95k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/1.39M [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/1.79k [00:00<?, ?B/s]

[{'summary_text': "Semarang is the capital of Central Java province in Indonesia and is known for its diverse culture and heritage, which reflect a blend of Javanese, Chinese, and Dutch influences. One of the city's most iconic landmarks is the Lawang Sewu, a historical building that once served as the headquarters of the Dutch East Indies Railway Company ."}]

### 8. Translate

In [ ]:
from transformers import pipeline

translator = pipeline("translation_id_to_en", model="Helsinki-NLP/opus-mt-id-en")

text_to_translate = "Hari ini aku kehujanan"
result = translator(text_to_translate)

print(result[0]['translation_text'])

config.json:   0%|          | 0.00/1.38k [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/291M [00:00<?, ?B/s]

generation_config.json:   0%|          | 0.00/293 [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/42.0 [00:00<?, ?B/s]

source.spm:   0%|          | 0.00/801k [00:00<?, ?B/s]

target.spm:   0%|          | 0.00/796k [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/1.26M [00:00<?, ?B/s]

/usr/local/lib/python3.10/dist-packages/transformers/models/marian/tokenization_marian.py:175: UserWarning: Recommended: pip install sacremoses.
  warnings.warn("Recommended: pip install sacremoses.")


Today I'm in the rain
